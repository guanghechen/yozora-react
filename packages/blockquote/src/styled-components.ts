import 'styled-components'
import type { YozoraBlockquoteTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    blockquote?: YozoraBlockquoteTheme
  }
}
