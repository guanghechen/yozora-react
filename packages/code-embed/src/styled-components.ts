import type { YozoraCodeEmbedTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeEmbed?: YozoraCodeEmbedTheme
  }
}
