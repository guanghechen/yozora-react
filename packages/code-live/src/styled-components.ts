import 'styled-components'
import type { YozoraCodeLiveTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    codeLive?: YozoraCodeLiveTheme
  }
}
