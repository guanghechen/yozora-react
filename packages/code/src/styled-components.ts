import 'styled-components'
import type { YozoraCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    code?: YozoraCodeTheme
  }
}
