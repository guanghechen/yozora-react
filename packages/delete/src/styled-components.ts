import type { YozoraDeleteTheme } from './theme'
import 'styled-components'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    delete?: YozoraDeleteTheme
  }
}
