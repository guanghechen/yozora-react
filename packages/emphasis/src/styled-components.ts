import 'styled-components'
import type { YozoraEmphasisTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    emphasis?: YozoraEmphasisTheme
  }
}
