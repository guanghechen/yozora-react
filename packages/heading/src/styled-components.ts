import 'styled-components'
import type { YozoraHeadingTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    heading?: YozoraHeadingTheme
  }
}
