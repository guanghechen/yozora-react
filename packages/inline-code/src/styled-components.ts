import 'styled-components'
import type { YozoraInlineCodeTheme } from './theme'


declare module 'styled-components' {
  export interface DefaultTheme {
    yozora: YozoraTheme
  }

  interface YozoraTheme {
    inlineCode: YozoraInlineCodeTheme
  }
}
