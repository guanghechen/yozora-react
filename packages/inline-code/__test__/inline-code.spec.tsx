import React from 'react'
import { mount, render } from 'enzyme'
import { DefaultTheme, ThemeProvider } from 'styled-components'
import InlineCode from '../src'


describe('basic rendering case', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('render a text value', () => {
    const text = 'Hello, world!'
    const wrapper = render(
      <InlineCode value={ text } />
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-text'
    const wrapper = render(
      <InlineCode className={ className } value={ text } />
    )
    expect(wrapper.hasClass(className)).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('value is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<InlineCode value={ value } />)
      }).toThrow(/Failed prop type: The prop `value` is marked as required/i)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLSpanElement>()
    const wrapper = mount(
      <InlineCode ref={ ref } data-value="waw" value="" />
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <InlineCode
        value="Hello, world!"
        style={{ color: 'orange', fontSize: '16px' }}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('snapshot with theme', () => {
    const theme: DefaultTheme = {
      yozora: {
        inlineCode: {
          padding: '2px',
          borderRadius: '3px',
          margin: '0 2px',
          background: 'hsla(210deg, 13%, 12%, 0.05)',
          lineHeight: 1.5,
          color: '#d81848',
          fontFamily: 'Consolas, monospace, sans-serif',
          fontSize: '1em',
          fontWeight: 'inherit',
          fontStyle: undefined,
          // whiteSpace: undefined,
        }
      }
    }

    const wrapper = mount(
      <ThemeProvider theme={ theme }>
        <InlineCode
          value="Hello, world!"
          style={{ color: 'orange', fontSize: '16px' }}
        />
      </ThemeProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
