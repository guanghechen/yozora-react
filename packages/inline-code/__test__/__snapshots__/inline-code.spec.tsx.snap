// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c0 {
  padding: var(--md-inline-code-padding,0.2em);
  border-radius: var(--md-inline-code-border-radius,2px);
  margin: var(--md-inline-code-margin,0);
  background: var(--md-inline-code-bg-color);
  color: var(--md-inline-code-color);
  font-family: var(--md-inline-code-font-family,'Consolas, monospace, sans-serif');
  font-size: var(--md-inline-code-font-size,1em);
  font-weight: var(--md-inline-code-font-weight,500);
  line-height: var(--md-inline-code-line-height,1.375);
  white-space: var(--md-inline-code-white-space,normal);
}

<InlineCode
  style={
    Object {
      "color": "orange",
      "fontSize": "16px",
    }
  }
  value="Hello, world!"
>
  <styled.code
    style={
      Object {
        "color": "orange",
        "fontSize": "16px",
      }
    }
  >
    <code
      className="c0"
      style={
        Object {
          "color": "orange",
          "fontSize": "16px",
        }
      }
    >
      Hello, world!
    </code>
  </styled.code>
</InlineCode>
`;
